pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -813,13 +817,16 @@ func applyConnectionPool(env *model.Environment, cluster *apiv2.Cluster, setting
 
 		if settings.Tcp.MaxConnections > 0 {
 			threshold.MaxConnections = &wrappers.UInt32Value{Value: uint32(settings.Tcp.MaxConnections)}
+			enableCircuitBreakers = true
 		}
 
 		applyTCPKeepalive(env, cluster, settings)
 	}
 
-	cluster.CircuitBreakers = &v2Cluster.CircuitBreakers{
-		Thresholds: []*v2Cluster.CircuitBreakers_Thresholds{threshold},
+	if enableCircuitBreakers {
+		cluster.CircuitBreakers = &v2Cluster.CircuitBreakers{
+			Thresholds: []*v2Cluster.CircuitBreakers_Thresholds{threshold},
+		}
```

Comment:
same applies to outlier detection as well doesn't it?

Reply:
yeah, this is envoy naming vs industry standards I think. my understanding was that what is causing the problems are the Envoy circuit breaker settings (which is a bit misleading because they define connection pool sizes), not the outlier detection settings. the problem being that in a big mesh, 1024 active retries per cluster is an arbitrarily small number which can cause problems when you have a lot of traffic.

@PiotrSikora please correct me if I'm wrong